test/goose/test_utils.clj

```diff
@@ -55,6 +58,71 @@
 
   (clear-redis))
 
+(def called? (atom false))
+(defn get-called? [] @called?)
+(defn set-called? [value] (reset! called? value))
+
+(defn reset-called?-fixture
+  [f]
+  (set-called? false)
+  (f))
+
+(defn stub [f]
+  (fn [& args]
+    (set-called? true)
+    (apply f args)))
+
+(defn job [overrides]
+  (merge (j/new `println (list "foobar") queue (d/prefix-queue queue) retry/default-opts)
+         overrides))
+
+(defn job-description [overrides]
+  (merge (j/description `println (list "foobar") queue (d/prefix-queue queue) retry/default-opts)
+         overrides))
+
+(defn cron-opts [overrides]
+  (merge {:cron-name     (str (random-uuid))
+          :cron-schedule "*/3 * * * *"
+          :timezone      "US/Pacific"} overrides))
+
+(defn enqueue-job [& [overrides]]
+  (let [j (job overrides)]
+    (redis-cmds/enqueue-back redis-conn (:ready-queue j) j)
+    (:id j)))
+
+(defn schedule-job [& [overrides]]
+  (let [{:keys [scheduled-at] :as j} (job (merge {:scheduled-at (u/epoch-time-ms)} overrides))]
+    (scheduler/run-at redis-conn scheduled-at j)
+    (:id j)))
+
+(defn periodic-job [& [overrides]]
+  (let [job-desc (job-description (:job-description overrides))
+        cron-opts (cron-opts (:cron-opts overrides))]
+    (cron/register redis-conn cron-opts job-desc)
+    cron-opts))
+
+(defn dead-job [& [overrides]]
+  (let [error-state {:state {:error           "Error"
+                             :last-retried-at (u/epoch-time-ms),
+                             :first-failed-at 1701344365468,
+                             :retry-count     27,
+                             :retry-at        1701344433359}}
+        j (job (merge error-state overrides))]
+    (redis-cmds/enqueue-sorted-set redis-conn d/prefixed-dead-queue
+                                   (get-in j [:state :last-retried-at]) j)
+    (:id j)))
+
+(defn add-jobs [{:keys [enqueued scheduled periodic dead]
+                 :or {enqueued 0 scheduled 0 periodic 0 dead 0}} & [overrides]]
+  (let [apply-fn-n-times (fn [n f & args]
+                 (dotimes [_ n] (apply f args)))]
+    (apply-fn-n-times enqueued enqueue-job (:enqueued overrides))
+    (apply-fn-n-times scheduled schedule-job (:scheduled overrides))
+    (apply-fn-n-times periodic periodic-job (:periodic overrides))
+    (apply-fn-n-times dead dead-job (:dead overrides))))
+
```

Comment:
These methods probably belong in a factory ns. Avoid adding things to `utils` unless there's no other place that it can belong.

It seems like `add-jobs` is only used in `console_test`, could start by keeping it there.